### PR TITLE
Use `workflow_triggers` table instead of an inline JSON

### PR DESCRIPTION
**Pull request by JanJakes**
## Description

Use `workflow_triggers` table instead of an inline JSON array that is not indexable.

## Code review notes

_N/A_

## QA notes

Automations only.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4778]

## After-merge notes

_N/A_

[MAILPOET-4778]: https://mailpoet.atlassian.net/browse/MAILPOET-4778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/Automation/Engine/Storage/WorkflowStorage.php [hunk 1]
```diff
@@ -211,4 +242,46 @@ private function insertWorkflowVersion(int $workflowId, Workflow $workflow): voi
...
+
+  private function insertWorkflowTriggers(int $workflowId, Workflow $workflow): void {
+    $triggerKeys = [];
+    foreach ($workflow->getSteps() as $step) {
+      if ($step->getType() === Step::TYPE_TRIGGER) {
+        $triggerKeys[] = $step->getKey();
+      }
+    }
+
+    if (!$triggerKeys) {
```
Comment (websupporter) [inline mailpoet/lib/Automation/Engine/Storage/WorkflowStorage.php]: Not sure, maybe we should delete existing trigger_keys when no trigger_keys are found. Just to have the correct representation in the database.

To reproduce: If you delete the trigger in your draft and store the draft, the old keys remain in the database.

File: mailpoet/lib/Automation/Engine/Storage/WorkflowStorage.php [hunk 2]
```diff
@@ -94,42 +105,48 @@ public function getWorkflows(array $status = null): array {
...
-    $triggerKeys = [];
-    foreach ($result as $item) {
-      /** @var string[] $keys */
-      $keys = Json::decode($item);
-      $triggerKeys = array_merge($triggerKeys, $keys);
-    }
-    return array_unique($triggerKeys);
+    $query = (string)$this->wpdb->prepare(
+      "
+        SELECT DISTINCT triggers.trigger_key
```
Comment (websupporter) [inline mailpoet/lib/Automation/Engine/Storage/WorkflowStorage.php]: `triggers` as a table alias seems quite close to the `CREATE TRIGGER` command. It gets strangely highlighted by my PHPStorm. I didnt find any reserved word `triggers` in MySql, but maybe we should either escape or rename the alias, just to be on the safe side?

File: mailpoet/lib/Automation/Engine/Storage/WorkflowStorage.php (hunk 1 above)
Comment (JanJakes) [reply mailpoet/lib/Automation/Engine/Storage/WorkflowStorage.php]: Oh, of course, totally missed that 🤦 Thanks!

File: mailpoet/lib/Automation/Engine/Storage/WorkflowStorage.php (hunk 2 above)
Comment (JanJakes) [reply mailpoet/lib/Automation/Engine/Storage/WorkflowStorage.php]: 👍 I'm also thinking about using short single-letter aliases. I think we're doing that in many parts of the MP codebase and it seems quite readable.

**Review by websupporter — APPROVED** [review]
lgtm

**Pull request merged**